File: Serene/Serene.Script/Imports/ServiceContracts/Meeting.MeetingAgendaRow.cs

namespace Serene.Meeting
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class MeetingAgendaRow
    {
        public Int32? MeetingAgendaId { get; set; }
        public Int32? AgendaTypeId { get; set; }
        public Int32? MeetingId { get; set; }
        public Int32? DisplayOrder { get; set; }
        public Int16? IsActive { get; set; }
        public Int32? RequestedByContactId { get; set; }
        public String Tags { get; set; }
        public String Title { get; set; }
        public String Description { get; set; }
        public Int32? ImageFileId { get; set; }
        public String AgendaTypeName { get; set; }
        public Int16? AgendaTypeIsActive { get; set; }
        public String MeetingMeetingGuid { get; set; }
        public String MeetingMeetingName { get; set; }
        public Int32? MeetingMeetingTypeId { get; set; }
        public String MeetingStartDate { get; set; }
        public String MeetingEndDate { get; set; }
        public Int32? MeetingInsertUserId { get; set; }
        public Int32? MeetingInsertDate { get; set; }
        public Int16? MeetingIsActive { get; set; }
        public Int32? MeetingUpdateUserId { get; set; }
        public Int32? MeetingUpdateDate { get; set; }
        public Int32? MeetingLocationId { get; set; }
        public Int32? MeetingOrganizationId { get; set; }
        public Int32? MeetingOrganizerContactId { get; set; }
        public Int32? MeetingReporterContactId { get; set; }
        public String RequestedByContactDisplayName { get; set; }
        public String RequestedByContactEmail { get; set; }
        public Int16? RequestedByContactIsActive { get; set; }
        public String RequestedByContactTitle { get; set; }
        public String RequestedByContactFirstname { get; set; }
        public String RequestedByContactLastname { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string MeetingAgendaId = "MeetingAgendaId";
            [InlineConstant] public const string AgendaTypeId = "AgendaTypeId";
            [InlineConstant] public const string MeetingId = "MeetingId";
            [InlineConstant] public const string DisplayOrder = "DisplayOrder";
            [InlineConstant] public const string IsActive = "IsActive";
            [InlineConstant] public const string RequestedByContactId = "RequestedByContactId";
            [InlineConstant] public const string Tags = "Tags";
            [InlineConstant] public const string Title = "Title";
            [InlineConstant] public const string Description = "Description";
            [InlineConstant] public const string ImageFileId = "ImageFileId";
            [InlineConstant] public const string AgendaTypeName = "AgendaTypeName";
            [InlineConstant] public const string AgendaTypeIsActive = "AgendaTypeIsActive";
            [InlineConstant] public const string MeetingMeetingGuid = "MeetingMeetingGuid";
            [InlineConstant] public const string MeetingMeetingName = "MeetingMeetingName";
            [InlineConstant] public const string MeetingMeetingTypeId = "MeetingMeetingTypeId";
            [InlineConstant] public const string MeetingStartDate = "MeetingStartDate";
            [InlineConstant] public const string MeetingEndDate = "MeetingEndDate";
            [InlineConstant] public const string MeetingInsertUserId = "MeetingInsertUserId";
            [InlineConstant] public const string MeetingInsertDate = "MeetingInsertDate";
            [InlineConstant] public const string MeetingIsActive = "MeetingIsActive";
            [InlineConstant] public const string MeetingUpdateUserId = "MeetingUpdateUserId";
            [InlineConstant] public const string MeetingUpdateDate = "MeetingUpdateDate";
            [InlineConstant] public const string MeetingLocationId = "MeetingLocationId";
            [InlineConstant] public const string MeetingOrganizationId = "MeetingOrganizationId";
            [InlineConstant] public const string MeetingOrganizerContactId = "MeetingOrganizerContactId";
            [InlineConstant] public const string MeetingReporterContactId = "MeetingReporterContactId";
            [InlineConstant] public const string RequestedByContactDisplayName = "RequestedByContactDisplayName";
            [InlineConstant] public const string RequestedByContactEmail = "RequestedByContactEmail";
            [InlineConstant] public const string RequestedByContactIsActive = "RequestedByContactIsActive";
            [InlineConstant] public const string RequestedByContactTitle = "RequestedByContactTitle";
            [InlineConstant] public const string RequestedByContactFirstname = "RequestedByContactFirstname";
            [InlineConstant] public const string RequestedByContactLastname = "RequestedByContactLastname";
        }
    }
    
}

File: Serene/Serene.Web/Imports/ServerTypings/Meeting.MeetingAgendaRow.ts
namespace Serene.Meeting {
    export interface MeetingAgendaRow {
        AgendaId?: number;
        MeetingId?: number;
        AgendaNumber?: number;
        Title?: string;
        Description?: string;
        AgendaTypeId?: number;
        RequestedByContactId?: number;
        Images?: string;
        Attachments?: string;
        MeetingMeetingName?: string;
        MeetingMeetingNumber?: string;
        MeetingMeetingGuid?: string;
        MeetingMeetingTypeId?: number;
        MeetingStartDate?: string;
        MeetingEndDate?: string;
        MeetingLocationId?: number;
        MeetingUnitId?: number;
        MeetingOrganizerContactId?: number;
        MeetingReporterContactId?: number;
        MeetingInsertUserId?: number;
        MeetingInsertDate?: string;
        MeetingUpdateUserId?: number;
        MeetingUpdateDate?: string;
        AgendaTypeName?: string;
        RequestedByContactTitle?: string;
        RequestedByContactFirstName?: string;
        RequestedByContactLastName?: string;
        RequestedByContactFullName?: string;
        RequestedByContactEmail?: string;
        RequestedByContactIdentityNo?: string;
        RequestedByContactUserId?: number;
    }

    export namespace MeetingAgendaRow {
        export const idProperty = 'AgendaId';
        export const nameProperty = 'Title';
        export const localTextPrefix = 'Meeting.MeetingAgenda';

        export namespace Fields {
            export declare const AgendaId: string;
            export declare const MeetingId: string;
            export declare const AgendaNumber: string;
            export declare const Title: string;
            export declare const Description: string;
            export declare const AgendaTypeId: string;
            export declare const RequestedByContactId: string;
            export declare const Images: string;
            export declare const Attachments: string;
            export declare const MeetingMeetingName: string;
            export declare const MeetingMeetingNumber: string;
            export declare const MeetingMeetingGuid: string;
            export declare const MeetingMeetingTypeId: string;
            export declare const MeetingStartDate: string;
            export declare const MeetingEndDate: string;
            export declare const MeetingLocationId: string;
            export declare const MeetingUnitId: string;
            export declare const MeetingOrganizerContactId: string;
            export declare const MeetingReporterContactId: string;
            export declare const MeetingInsertUserId: string;
            export declare const MeetingInsertDate: string;
            export declare const MeetingUpdateUserId: string;
            export declare const MeetingUpdateDate: string;
            export declare const AgendaTypeName: string;
            export declare const RequestedByContactTitle: string;
            export declare const RequestedByContactFirstName: string;
            export declare const RequestedByContactLastName: string;
            export declare const RequestedByContactFullName: string;
            export declare const RequestedByContactEmail: string;
            export declare const RequestedByContactIdentityNo: string;
            export declare const RequestedByContactUserId: string;
        }

        ['AgendaId', 'MeetingId', 'AgendaNumber', 'Title', 'Description', 'AgendaTypeId', 'RequestedByContactId', 'Images', 'Attachments', 'MeetingMeetingName', 'MeetingMeetingNumber', 'MeetingMeetingGuid', 'MeetingMeetingTypeId', 'MeetingStartDate', 'MeetingEndDate', 'MeetingLocationId', 'MeetingUnitId', 'MeetingOrganizerContactId', 'MeetingReporterContactId', 'MeetingInsertUserId', 'MeetingInsertDate', 'MeetingUpdateUserId', 'MeetingUpdateDate', 'AgendaTypeName', 'RequestedByContactTitle', 'RequestedByContactFirstName', 'RequestedByContactLastName', 'RequestedByContactFullName', 'RequestedByContactEmail', 'RequestedByContactIdentityNo', 'RequestedByContactUserId'].forEach(x => (<any>Fields)[x] = x);
    }
}

